ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncDownloadPipelineFactory.java

```diff
@@ -194,4 +199,10 @@ protected boolean shouldContinueDownloadingFromPeer(
     }
     return shouldContinue;
   }
+
+  private long getPreMergeHeaderBlockNumber(final SyncState syncState) {
+    return syncConfig.isSnapSyncSavePreMergeHeadersOnlyEnabled()
+        ? syncState.getCheckpoint().map(checkpoint -> checkpoint.blockNumber() - 1).orElse(0L)
```

Comment:
Potentially a breaking change if any private networks make use of the existing checkpoint config? 

We're kind of overloading the checkpoint concept, which I think makes sense for public networks as long as we're explicit about it. We should check usage for private.
In full 4444s, with rolling window, I guess this static checkpoint goes away entirely for public nets?